src/modules/site-v2/templates/_includes/phenotype-database-table.html

```diff
@@ -270,27 +270,32 @@ <h1 class="h5" id="offcanvas1Label"></h5>
             }).get()
         }
         const data = {csrf_token: $('#csrf_token').val(), search_val: $("#traitsSearch").val(), selected_tags: selectedTags}
-        await sendPostRequest(data)
-            .then( resp => handlePostRequest(resp))
-            .catch( e => {
-                console.error(e)
-                flash_message(`Failed to retrieve the list of traits: ${e.statusText}`)
-            })
+        try {
+            const res = await sendPostRequest(data)
+            return handlePostRequest(res)
+        } catch(err) {
+            console.error(err)
+            flash_message(`Failed to retrieve the list of traits: ${err.statusText}`)
+        }
     }
 
-    function sendPostRequest(data) {
-        return new Promise((resolve, reject) => {
-            $.ajax({
-                type: "POST",
-                url: "{{ url_for('phenotype_database.phenotype_database') }}",
-                data: JSON.stringify(data),
-                contentType: "application/json",
-                dataType: "json",
-                success: (response) => resolve(response),
-                error: (error)=> reject(error)
-            })
+    async function sendPostRequest(data) {
+        const resp = await fetch("{{ url_for('phenotype_database.phenotype_database') }}", {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+                'X-CSRF-TOKEN': data.csrf_token
+            },
+            body: JSON.stringify(data)
         })
-                
+
+        if (resp.ok) {
+            const respData = await resp.json()
```

Comment:
if the server returns html instead of valid json, this call (`await res.json()`) will also throw an exception and will need  to be wrapped into try/catch. You might combine both the `await fetch()` and `await res.json()` int he same try/catch block, if it makes sense.